app/src/main/java/com/doubtless/doubtless/screens/answers/usecases/PublishAnswerUseCase.kt

```diff
@@ -10,21 +10,32 @@ class PublishAnswerUseCase constructor(
     private val doubtlessServer: DoubtlessServer
 ) {
 
+    interface PublishAnswerListener {
+        fun onPublishAnswerLoading(isLoading: Boolean)
+    }
+
     sealed class Result {
         class Success(val answerData: AnswerData): Result()
         class Error(val message: String): Result()
     }
 
-    suspend fun publish(publishAnswerRequest: PublishAnswerRequest) = withContext(Dispatchers.IO) {
+    suspend fun publish(publishAnswerRequest: PublishAnswerRequest,
+                        listener: PublishAnswerListener? = null) = withContext(Dispatchers.IO) {
         // centralise error handling
         return@withContext try {
 
+            // Notify loading state to the listener
+            listener?.onPublishAnswerLoading(true)
```

Comment:
didn't get it. what's the use of getting loading state from usecase, the viewmodel can directly know if its loading or not.